pkg/dataplane/client.go

```diff
@@ -1,19 +1,47 @@
 package dataplane
 
 import (
+	"context"
+	"fmt"
+	"reflect"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/msi-dataplane/internal/swagger"
+	"github.com/go-playground/validator/v10"
 )
 
+//go:generate /bin/bash -c "../../hack/mockgen.sh mock_swagger_client/zz_generated_mocks.go client.go"
+
 // TODO - Tie the module version to update automatically with new releases
 const moduleVersion = "v0.0.1"
 
 type ManagedIdentityClient struct {
-	swaggerClient *swagger.ManagedIdentityDataPlaneAPIClient
+	swaggerClient swaggerMSIClient
 }
 
+type UserAssignedMSIRequest struct {
+	IdentityURL string `validate:"required,url"`
+	ResourceID  string `validate:"required"`
+	TenantID    string `validate:"required,uuid"`
+}
+
+type swaggerMSIClient interface {
+	Getcreds(ctx context.Context, credRequest swagger.CredRequestDefinition, options *swagger.ManagedIdentityDataPlaneAPIClientGetcredsOptions) (swagger.ManagedIdentityDataPlaneAPIClientGetcredsResponse, error)
+}
+
+var _ swaggerMSIClient = &swagger.ManagedIdentityDataPlaneAPIClient{}
+
+var (
+	errGetCreds           = fmt.Errorf("failed to get credentials")
+	errInvalidRequest     = fmt.Errorf("invalid request")
+	errNilField           = fmt.Errorf("expected non-nil field in user-assigned managed identity")
+	errNilMSI             = fmt.Errorf("expected non-nil user-assigned managed identity")
+	errNotOneMSI          = fmt.Errorf("expected one user-assigned managed identity")
```

Comment:
These are almost the same - should they be combined?

Reply:
I say let's keep them separate. In the future, if we have scenarios where we want to assign multiple user-assigned managed identities, we wouldn't use `errNotOneMSI`.